Source/WebGPU/WGSL/GlobalVariableRewriter.cpp

```diff
@@ -729,22 +753,47 @@ const Type* RewriteGlobalVariables::packArrayType(const Types::Array* arrayType)
     return m_callGraph.ast().types().arrayType(packedStructType, arrayType->size);
 }
 
-void RewriteGlobalVariables::visitEntryPoint(AST::Function& function, AST::StageAttribute::Stage stage, PipelineLayout& pipelineLayout)
+void RewriteGlobalVariables::visitEntryPoint(const CallGraph::EntryPoint& entryPoint)
 {
     m_reads.clear();
     m_structTypes.clear();
 
-    dataLogLnIf(shouldLogGlobalVariableRewriting, "> Visiting entrypoint: ", function.name());
+    dataLogLnIf(shouldLogGlobalVariableRewriting, "> Visiting entrypoint: ", entryPoint.function.name());
+
+    m_entryPointInformation = &entryPoint.information;
+
+    auto it = m_pipelineLayouts.find(m_entryPointInformation->originalName);
+    ASSERT(it != m_pipelineLayouts.end());
+
+    if (!it->value.has_value()) {
+        m_entryPointInformation->defaultLayout = { PipelineLayout { } };
+        m_generatedLayout = &*m_entryPointInformation->defaultLayout;
+    } else
+        m_generatedLayout = nullptr;
 
-    visit(function);
+    switch (entryPoint.stage) {
+    case AST::StageAttribute::Stage::Compute:
+        m_stage = ShaderStage::Compute;
```

Comment:
Wouldn't it be 😎 if these were the same?

Reply:
Yeah, I've been slowly unifying these enums, I'll get to more of them soon...